app/services/BigQueryService.scala

```diff
@@ -0,0 +1,119 @@
+package services
+
+import com.google.api.gax.core.FixedCredentialsProvider
+import com.google.auth.oauth2.GoogleCredentials
+import com.google.cloud.RetryOption
+import com.google.cloud.bigquery.{BigQuery, BigQueryError, FieldValueList, JobInfo, QueryJobConfiguration, TableResult}
+import com.typesafe.scalalogging.LazyLogging
+import models.BigQueryResult
+import zio.blocking.{Blocking, effectBlocking}
+import zio.{ZIO}
+
+import java.io.ByteArrayInputStream
+import scala.jdk.CollectionConverters._
+
+class BigQueryService(bigQuery: BigQuery) extends LazyLogging {
+  def buildQuery(testName: String, channel:String, stage: String): String = {
+
+    val channelInQuery = channel match {
+      case "Epic" => "ACQUISITIONS_EPIC"
+      case "Banner1" =>"ACQUISITIONS_ENGAGEMENT_BANNER"
+      case "Banner2" =>"ACQUISITIONS_SUBSCRIPTIONS_BANNER"
+    }
+
+    s"""WITH ltv3dataForTest AS (SELECT
+      ab.name AS test_name,
+      ab.variant AS variant_name,
+      component_type,
+      SUM(acquisition_ltv_3_year) AS ltv3,
+    FROM `datatech-platform-prod.reader_revenue.fact_holding_acquisition`
+    CROSS JOIN UNNEST(ab_tests) AS ab
+    WHERE  ab.name= '$testName'
+    AND component_type ='$channelInQuery'
+    GROUP BY 1,2,3
+    )
+    SELECT test_name, variant_name,component_type,ltv3 FROM ltv3dataForTest
+    """
+  }
+
+  def runQuery(queryString: String) : zio.RIO[Blocking, Either[BigQueryError, TableResult]] = {
```

Comment:
It's preferable to let zio model the error rather than nesting an Either, e.g.
`ZIO[ZEnv, BigQueryError, TableResult]`

So instead of creating `Right` and `Left` values, you can instead call:
`ZIO.succeed(...)` and `ZIO.fail(...)`

For an example, see https://github.com/guardian/support-admin-console/blob/main/app/services/DynamoChannelTests.scala#L55